.github/workflows/ci.yml

```diff
@@ -73,7 +73,7 @@ jobs:
       # We could run 'make lint' but we prefer this action because it leaves
       # 'annotations' (i.e. it comments on PRs to point out linter violations).
       - name: Lint
-        uses: golangci/golangci-lint-action@v3
+        uses: golangci/golangci-lint-action@v7
```

Comment:
There is an issue related to `golangci-lint-action`. It seems some fields are not supported in workflow anymore so the workflow keep reporting a failure on the lint step. Bump it up to v7 to avoid, and golangci to v2.